### ISSUE TITLE
cannot force contentType in addPage

### ISSUE THREAD

**Issue by Naouak** [mikehaertl/phpwkhtmltopdf#169]
When using addPage, the system uses a regex to determine what type of content the input is. If your content is malformed but still valid for webkit, it will use the input as a filename instead of saving it to a file.

addPage needs a way to tell whether it' s a filename or content given.

**Example :**
````php
$pdf->addPage("<h1>TEST</h1>");
````
Will try to render the file "\<h1\>TEST\</h1\>" instead of rendering HTML.

**Comment by mikehaertl** [comment]
That's correct and there's not really a simple way to fix this. So you should make sure, that your HTML strings always contain a `<html>` tag somewhere.

**Comment by Naouak** [comment]
It's simple to fix this, permit to pass an override to the function. Here is a gist how I could do it easily (which could be easily backported to lib):
https://gist.github.com/Naouak/98b2afc88386dbfafc4d97c09f86be49

**Comment by mikehaertl** [comment]
I'm not sure, what you try to achieve. How is this related to the content type? Can't you simply set that in your HTML string?

**Comment by Naouak** [comment]
I'm building a middleware for rendering a webpage in pdf in a framework. I don't know what kind of html I will receive. Webkit can perfectly parse some HTML without the doctype or even the <html> tag, I want to be able to have a pdf when those are absent without checking myself first. It's stupid to run twice the regexp when I already know what it's supposed to be.

**Comment by mikehaertl** [comment]
Ok, seems to make sense. I've added your suggestion to `master`. Note, that you should use `Pdf::TYPE_HTML`. Could you help testing?

**Comment by Naouak** [comment]
Thanks. Sorry I couldn't help you test it.

**Comment by mikehaertl** [comment]
Feedback is still welcome, so it would be great to let me know if the change now works for you.
